### PR TITLE
Document overflow behavior, add tests.

### PR DESCRIPTION
**Pull request by JimLarson**

### CODE REVIEW

**Comment by JimLarson** [comment]
Closes #131

File: doc/langdef.md [hunk 1]
```diff
@@ -827,6 +827,12 @@ values. The ordering operators obey the usual algebraic properties, i.e. `e1 <=
 e2` gives the same result as `!(e1 > e2)` as well as `(e1 < e2) || (e1 == e2)`
 when the expressions involved do not have side effects.
 
+### Overflow
```
Comment (TristonianJones) [inline doc/langdef.md:830]: Could we add a test to ensure negating the minimum `int` value will overflow?

Comment (JimLarson) [reply doc/langdef.md:830]: Done. Thanks, good catch. In cel-go it wraps back to itself, but for consistency we should expect an error.

**Comment by JimLarson** [comment]
PTAL

File: tests/simple/testdata/integer_math.textproto [hunk 2]
```diff
@@ -200,8 +198,14 @@ section {
     eval_error: {
       errors: { message: "return error for overflow" }
     }
-    ## Current behavior:
-    ## value: { int64_value: 9223372036854775807 }
+  }
+  test {
+    name: "int64_min_negate"
+    description: "Negated LLONG_MIN is not representable."
```
Comment (TristonianJones) [inline tests/simple/testdata/integer_math.textproto:204]: nit: extra `L` in `LONG`?

Comment (JimLarson) [reply tests/simple/testdata/integer_math.textproto:204]: I think Gabriel was using standard C constants, where "long" might only be 32-bits, so it's for "long long min", guaranteeing at least 64 bits. Long long could be bigger than that, but C's flexibility for integer sizes is mostly theoretical these days - at least outside the embedded market.
